From 2252b1d98f6454ff7840c8aff41a21baed3646bb Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@intel.com>
Date: Fri, 8 May 2015 17:51:16 +0300
Subject: [PATCH 04/11] iio: hrtimer: Avoid using CONFIGFS defines

This makes code easier to read.

Change-Id: I442289672412e543bbfa079a44c39d05a612af9a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26476
Signed-off-by: Daniel Baluta <daniel.baluta@intel.com>
Reviewed-on: https://android.intel.com:443/486297
---
 drivers/iio/trigger/iio-trig-hrtimer.c |  7 +++----
 include/linux/iio/sw_trigger.h         | 14 ++++++++++++++
 2 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/drivers/iio/trigger/iio-trig-hrtimer.c b/drivers/iio/trigger/iio-trig-hrtimer.c
index 33fe3f5..698258d 100644
--- a/drivers/iio/trigger/iio-trig-hrtimer.c
+++ b/drivers/iio/trigger/iio-trig-hrtimer.c
@@ -154,10 +154,9 @@ static struct iio_sw_trigger *iio_trig_hrtimer_probe(const char *name)
 	ret = iio_trigger_register(trig_info->swt.trigger);
 	if (ret)
 		goto err_free_trigger;
-#ifdef CONFIG_CONFIGFS_FS
-	config_group_init_type_name(&trig_info->swt.group, name,
-				    &iio_hrtimer_type);
-#endif
+
+	iio_config_group_init_type_name(&trig_info->swt.group, name,
+					&iio_hrtimer_type);
 	return &trig_info->swt;
 err_free_trigger:
 	iio_trigger_free(trig_info->swt.trigger);
diff --git a/include/linux/iio/sw_trigger.h b/include/linux/iio/sw_trigger.h
index ebc4dc6..87803c7 100644
--- a/include/linux/iio/sw_trigger.h
+++ b/include/linux/iio/sw_trigger.h
@@ -66,6 +66,20 @@ struct iio_sw_trigger *to_iio_sw_trigger(struct config_item *item)
 	return container_of(to_config_group(item), struct iio_sw_trigger,
 			    group);
 }
+
+static inline
+void iio_config_group_init_type_name(struct config_group *group,
+				     const char *name,
+				     struct config_item_type *type)
+{
+	config_group_init_type_name(group, name, type);
+}
+#else
+static inline
+void iio_config_group_init_type_name(struct config_group *group,
+				     const char *name,
+				     struct config_item_type *type)
+{ }
 #endif
 
 #endif /* __IIO_SW_TRIGGER */
-- 
1.9.1

